### ISSUE TITLE
Provider "mongodbatlas" not available for installation.

### ISSUE THREAD

**Issue by abramovi** [akshaykarle/terraform-provider-mongodbatlas#73]
Hi ,

I am trying to use this provider following the example folder, but when I ran terraform init, i am getting - Provider "mongodbatlas" not available for installation. 

Any idea ?

**Comment by Knappek** [comment]
@abramovi have you installed the provider? Check the Readme for installation guide, it's basically about downloading the binary from the release page and putting it into ~/.terraform.d/plugins

<sub>Sent with <a href="http://githawk.com">GitHawk</a></sub>

**Comment by gschaffer-cxn** [comment]
We are experiencing the same issues. Downloaded the binary, put in into ~/.terraform.d/plugins and .../pluginslinux_amd64/ and get the same error. :(

**Comment by Knappek** [comment]
I'm currently not on my computer so I can't reproduce it, but I guess the filename is not correct, you have to remove the "_linux_amd64" according to the naming scheme: https://www.terraform.io/docs/configuration/providers.html#third-party-plugins

<sub>Sent with <a href="http://githawk.com">GitHawk</a></sub>

**Comment by christianuhlcc** [comment]
Hi @Knappek we just experienced this issue and renaming (removing  "_linux_amd64" ) helps, plugin is installed correctly

**Comment by gschaffer-cxn** [comment]
That fixed it for me aswell, thanks

**Comment by Knappek** [comment]
@akshaykarle you can close this issue.

<sub>Sent with <a href="http://githawk.com">GitHawk</a></sub>

**Comment by christianuhlcc** [comment]
@Knappek I would leave it open for a follow-up PR that fixes the actual issue - that the name of the compiled binary is not suitable for installation, otherwise every new user needs to find this closed issue to install it, right?

 I can try to propose a PR, but I am horrible at go, so I for sure need some help.

**Comment by Knappek** [comment]
@christianuhlcc I don't think this is possible as you need to differentiate between the different operating systems...it's quite common that you need to rename binaries that you download from the web. What you can do though is enhancing the Readme maybe...

<sub>Sent with <a href="http://githawk.com">GitHawk</a></sub>

**Comment by gschaffer-cxn** [comment]
@Knappek the same issue exists within the zipped releases, we could at least fix those along with the README

**Comment by christianuhlcc** [comment]
I looked around how other community providers handle it, and everybody does it differently ... 

Maybe only serving ZIP files with the target system in the name, and the containing file in the usable filename would be the way to go? That would make it obvious for downloaders and easy for installers to do the right thing.

**Comment by akshaykarle** [comment]
hey all, thanks for all the inputs. I will update the readme with instructions to rename the binaries to avoid confusion.

**Comment by nitrag** [comment]
This was my first 3rd party plugin, phew complicated.

```
wget -O ~/.terraform.d/plugins/terraform-provider-mongodbatlas_v1.1.0 https://github.com/akshaykarle/terraform-provider-mongodbatlas/releases/download/v1.1.0/terraform-provider-mongodbatlas_v1.1.0_darwin_amd64
```

**Comment by petr-nazarov** [comment]
This worked for me, adding the following lines before using any resources:
```
terraform {
  required_providers {
    mongodbatlas = {
      source  = "mongodb/mongodbatlas"
      version = "0.9.1"
    }
  }
}

provider "mongodbatlas" {
  public_key  = var.mongodb_atlass_public_key
  private_key = var.mongodb_atlass_private_key
}
```
